lib/cdnetworks-client.rb

```diff
@@ -21,15 +31,20 @@ def compose_request(path,options)
     request
   end
 
-  def call(path,options)
-    begin
-      response = http.request(compose_request(path,options))
+  def call(path,options,session_retries=0)
+    response = http.request(compose_request(path,options))
+
+    if @location == "Beta"
+      process_beta_response(path, options, response, session_retries)
+    else
       response_hash = { code: response.code, body: response.body }
-    rescue StandardError=>e
-      "An error has occurred connecting to the CDNetworks API (#{e})"
     end
   end
 
+  def location
+    @location
```

Comment:
I'd request that we make use of the assignment methods provided by the attr_accessor method generator as opposed to operating on the instance variables directly. It's a small thing but a stylistic abstraction that we've landed on. In the initializer, `@location = credentials[:location]`, could then be `self.location = credentials[:location]`. The same goes for `@user` and `@password`